test/integration/integration_test.go

```diff
@@ -265,6 +265,25 @@ func TestUnregisteringClusterServiceBroker(t *testing.T) {
 	suite.waitForClusterServiceBrokerNotRegistered()
 }
 
+func TestRetryOnNetworkError(t *testing.T) {
```

Comment:
I am not sure if this test works properly but when I change the error returned on `OnAdd` method from temporary to regular, the test still passes. Is this desirable behavior?

Reply:
Returning regular error makes retry without any delay. Removing the `return` (which makes the code like before the PR) makes the test red.